### ISSUE TITLE
CLI version 3.47.2 breaks awsx.lb

### ISSUE THREAD

**Issue by MitchellGerdisch** [pulumi/pulumi#11450]
### What happened?

With CLI 3.47.1 the code below works. 
Upgrading to CLI 3.47.2, it breaks with this error

```
aws:lb:TargetGroup (mitch-nginx-lb):
    error: resource urn:pulumi:dev::bkv-ecs-issue::awsx:lb:ApplicationLoadBalancer$aws:lb/targetGroup:TargetGroup::mitch-nginx-lb violates plan: properties changed: ++name[{mitch-nginx-lb-b7c1331}!={mitch-nginx-lb-628476a}]
```

```
import pulumi_awsx as awsx

name_base = "mitch"
lb = awsx.lb.ApplicationLoadBalancer(f"{name_base}-nginx-lb")
```

### Steps to reproduce

1. Install Pulumi cli 3.47.1 (or older)
2. `pulumi up` the stack given above
3. see no error
4. `pulumi destroy` the stack
5. Upgrade to 3.47.2
6. `pulumi up` 
7. See the error given above

### Expected Behavior

Should work in 3.47.2

### Actual Behavior

Fails wiht 3.47.2

### Output of `pulumi about`

CLI          
Version      3.47.1
Go Version   go1.19.2
Go Compiler  gc

Plugins
NAME    VERSION
aws     5.21.1
awsx    1.0.0-beta.11
docker  3.6.1
python  unknown

Host     
OS       darwin
Version  12.5.1
Arch     x86_64

This project is written in python: executable='/Users/mitch/Downloads/bkv-ecs-issue/venv/bin/python3' version='3.9.6
'

Current Stack: dev

Found no resources associated with dev

Found no pending operations associated with dev

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by jaxxstorm** [comment]
We're having reports this also affects Helm Release as well

**Comment by justinvp** [comment]
Duplicate of https://github.com/pulumi/pulumi/issues/11444, which has been fixed in v3.48.0.
